### PR TITLE
Meta Winner Ensemble

### PR DESCRIPTION
**Pull request by marcharper**
New strategy: Meta Winner Ensemble

Rather than take the "best" strategy, this MetaWinner variant chooses randomly from some of the top strategies computed each round against each opponent. In the full tournament it comes in 7th and is now the current best meta strategy.

I spent a little time tuning the parameter of how many strategies to pick from but not a lot, so it's possible that there's room for improvement.

### CODE REVIEW

File: axelrod/strategies/meta.py [hunk 1]
```diff
@@ -165,6 +165,32 @@ def meta_strategy(self, results, opponent):
...
+        index = choice([i for (s, i) in scores[:prop]])
+
+        # Update each player's proposed history with his proposed result, but
+        # always after the new result has been settled based on scores
+        # accumulated until now.
+        for r, t in zip(results, self.team):
+            t.proposed_history.append(r)
+
+        if opponent.defections == 0:
+            # Don't poke the bear
```
Comment (meatballs) [inline axelrod/strategies/meta.py]: 😄

File: axelrod/strategies/meta.py [hunk 2]
```diff
@@ -165,6 +165,32 @@ def meta_strategy(self, results, opponent):
 
         return bestresult
 
+class MetaWinnerEnsemble(MetaWinner):
+    """A variant of MetaWinner that chooses one of the top scoring strategies at random against each opponent."""
```
Comment (drvinceknight) [inline axelrod/strategies/meta.py]: Could you add something like:

```
Names:

Meta Winner Ensemble: Original name by Marc Harper
```

File: axelrod/tests/unit/test_meta.py [hunk 3]
```diff
@@ -185,6 +185,27 @@ def test_strategy(self):
...
+        'manipulates_state': False
+    }
+
+    expected_class_classifier = copy.copy(expected_classifier)
+    expected_class_classifier['stochastic'] = False
+    expected_class_classifier['makes_use_of'] = set([])
+
+    def test_strategy(self):
+        self.first_play_test(C)
+
```
Comment (drvinceknight) [inline axelrod/tests/unit/test_meta.py]: Could you add a test (for a given random seed) just for a few plays please.

File: axelrod/strategies/meta.py (hunk 2 above)
Comment (marcharper) [inline axelrod/strategies/meta.py]: done

File: axelrod/tests/unit/test_meta.py (hunk 3 above)
Comment (marcharper) [inline axelrod/tests/unit/test_meta.py]: I added two deterministic test cases.

**Comment by marcharper** [comment]
I fixed the doc test and squashed the commit into the first one.

**Comment by drvinceknight** [comment]
Thanks @marcharper :)

**Pull request merged**